Source/WebCore/dom/DocumentFullscreen.cpp

```diff
@@ -375,7 +375,7 @@ static Vector<Ref<Document>> documentsToUnfullscreen(Frame& firstFrame)
         if (!document)
             continue;
         ASSERT(document->fullscreen().fullscreenElement());
-        if (!document->fullscreen().isSimpleFullscreenDocument())
+        if (!documents.isEmpty() && !documents.last()->fullscreen().isSimpleFullscreenDocument())
             break;
         if (RefPtr iframe = dynamicDowncast<HTMLIFrameElement>(document->ownerElement()); iframe && iframe->hasIFrameFullscreenFlag())
             break;
```

Comment:
Most of the algorithm is shifted by one document according to: https://fullscreen.spec.whatwg.org/#collect-documents-to-unfullscreen

So what's currently there isn't fully correct